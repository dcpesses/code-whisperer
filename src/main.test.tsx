import { vi, test, expect } from 'vitest';
// import {createRoot} from 'react-dom/client';
import {render, screen} from '@testing-library/react';
import Main from '@/main';

const { mockRender } = vi.hoisted(() => {
  return {
    mockRender: vi.fn()
  };
});

vi.mock('react-dom/client', () => ({
  createRoot: vi.fn().mockImplementation(() => ({
    render: mockRender,
  }))
}));

describe('Main', () => {
  test('should match the snapshot when the JSXElement is rendered as component', () => {
    const MainComponent = () => Main;

    const {container} = render(<MainComponent />);
    expect(container).toMatchSnapshot();

    const loadingElement = screen.getByText(/Loading/i);
    expect(loadingElement).toBeInTheDocument();
  });
});
