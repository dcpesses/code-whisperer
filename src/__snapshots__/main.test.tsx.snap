// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main > should match the snapshot when the JSXElement is rendered as component 1`] = `
<div>
  <div
    id="authenticated-app container text-center"
  >
    <div
      class="col display-6 text-center full-pg"
    >
      <div />
      <div>
        <div
          class="loading-ripple"
        >
          <div />
          <div />
        </div>
        <br />
        Loading...
      </div>
      <div />
    </div>
  </div>
</div>
`;
